generators/server/templates/build.gradle.ejs

```diff
@@ -542,6 +542,9 @@ dependencies {
     liquibaseRuntime "org.liquibase.ext:liquibase-mssql:<%= LIQUIBASE_VERSION %>"
     <%_ } _%>
     implementation "org.mapstruct:mapstruct:${mapstructVersion}"
+    <%_ if (reactive) { _%>
+    implementation "org.springframework.data:spring-data-commons"
+    <%_ } _%>
```

Comment:
IMO spring-data should not be required without a database.
Where it's used?

Reply:
In GatewaySwaggerResourcesProvider.java.ejs autowires SwaggerResources which is InMemorySwaggerResourcesProvider
Which Autoconfigure the Swagger Configuration where it uses the Pageable from 'org.springframework.data'

https://github.com/jhipster/jhipster/blob/main/jhipster-framework/src/main/java/io/github/jhipster/config/apidoc/SpringfoxAutoConfiguration.java


return docket
            .apiInfo(apiInfo)
            .useDefaultResponseMessages(properties.isUseDefaultResponseMessages())
            .groupName(MANAGEMENT_GROUP_NAME)
            .host(properties.getHost())
            .protocols(new HashSet<>(Arrays.asList(properties.getProtocols())))
            .forCodeGeneration(true)
            .directModelSubstitute(ByteBuffer.class, String.class)
            .genericModelSubstitutes(ResponseEntity.class)
            **.ignoredParameterTypes(Pageable.class)**
            .select()
            .paths(regex(managementContextPath + ".*"))
            .build();

Reply:
You are right.
https://github.com/jhipster/jhipster-bom/blob/3489f74b0e2d25b9e1e61a9352f798cfc42776b9/jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java#L172

Reply:
@yuvarajsanjeevi can you try to make Page conditional at jhipster-bom?
Something like:
```
        builder = docket
            .apiInfo(apiInfo)
            .useDefaultResponseMessages(properties.isUseDefaultResponseMessages())
            .groupName(MANAGEMENT_GROUP_NAME)
            .host(properties.getHost())
            .protocols(new HashSet<>(Arrays.asList(properties.getProtocols())))
            .forCodeGeneration(true)
            .directModelSubstitute(ByteBuffer.class, String.class)
            .genericModelSubstitutes(ResponseEntity.class)
        try {
          builder.ignoredParameterTypes(Class.forName("org.springframework.data.domain.Pageable"))
        } catch(error) {
        }
        return builder
            .select()
            .paths(regex(properties.getManagementIncludePattern()))
            .build();
```

Reply:
@mshima 
Created Pull request in jhipster bom
https://github.com/jhipster/jhipster-bom/pull/430

Reply:
Is possible to drop the dependency with this change?

Reply:
@yuvarajsanjeevi can you add the dependency again, please?
I’ve triggered the same build and didn’t work.
We probably should make syncLdp logic optional to drop spring-data dependency.

Reply:
@mshima Can you please share with me the build number or error details

Reply:
Same from https://github.com/jhipster/generator-jhipster/pull/15978#issuecomment-903045764
https://github.com/hipster-labs/jhipster-daily-builds/actions/runs/1152758979

Reply:
IMO we do not required ReactivePageableHandlerMethodArgumentResolver, ReactiveSortHandlerMethodArgumentResolver if databaseType is 'no'

Reply:
Can you please trigger the same build again

Reply:
done

Reply:
It worked.